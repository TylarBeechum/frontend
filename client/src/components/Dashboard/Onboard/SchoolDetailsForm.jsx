import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { MaskedInput, Select, Box, Heading } from 'grommet';

import {
  BaseForm,
  BaseTextInput,
  BaseFormField,
  BaseButton
} from '../../../styles/themes';
import queries from './queries';

const SchoolDetailsForm = ({ history }) => {
  const [input, setInput] = useState({
    name: '',
    taxId: '',
    street1: '',
    street2: '',
    city: '',
    state: '',
    zip: '',
    phone: '',
    type: '',
    url: '',
    userId: localStorage.id
  });

  const states = [
    'AL',
    'AK',
    'AS',
    'AZ',
    'AR',
    'CA',
    'CO',
    'CT',
    'DE',
    'DC',
    'FM',
    'FL',
    'GA',
    'GU',
    'HI',
    'ID',
    'IL',
    'IN',
    'IA',
    'KS',
    'KY',
    'LA',
    'ME',
    'MH',
    'MD',
    'MA',
    'MI',
    'MN',
    'MS',
    'MO',
    'MT',
    'NE',
    'NV',
    'NH',
    'NJ',
    'NM',
    'NY',
    'NC',
    'ND',
    'MP',
    'OH',
    'OK',
    'OR',
    'PW',
    'PA',
    'PR',
    'RI',
    'SC',
    'SD',
    'TN',
    'TX',
    'UT',
    'VT',
    'VI',
    'VA',
    'WA',
    'WV',
    'WI',
    'WY'
  ];

  const handleChanges = e => {
    setInput({
      ...input,
      [e.target.name]: e.target.value
    });
  };

  const handleSubmit = async e => {
    e.preventDefault();
    try {
      await queries.addSchoolDetails(input);
      await queries.addRole({
        id: localStorage.id,
        roleId: 2 // Role of a school is set to always be 2
      });
      history.push('/dashboard');
    } catch (err) {
      console.error(err);
    }
  };

  return (
    <SchoolForm onSubmit={handleSubmit}>
      <Box direction="column">
        <Heading margin="20px 0 0 0" alignSelf="center">
          School Details
        </Heading>
        <SchoolFormField label="Institution">
          <SchoolBaseTextInput
            name="name"
            placeholder="Lambda School"
            onChange={handleChanges}
            value={input.name}
            plain={false}
            required
          />
        </SchoolFormField>
        <SchoolFormField label="Tax Id">
          <SchoolBaseTextInput
            labelText="TaxId"
            name="taxId"
            placeholder="TaxId"
            onChange={handleChanges}
            value={input.taxId}
            required
          />
        </SchoolFormField>
        <SchoolFormField label="Address 1">
          <SchoolBaseTextInput
            name="street1"
            placeholder="123 Fake street"
            onChange={handleChanges}
            value={input.street1}
          />
        </SchoolFormField>
        <SchoolFormField label="Address 2">
          <SchoolBaseTextInput
            name="street2"
            placeholder="Apt B"
            onChange={handleChanges}
            value={input.street2}
          />
        </SchoolFormField>
        <SchoolFormField label="City">
          <SchoolBaseTextInput
            name="city"
            placeholder="San Francisco"
            onChange={handleChanges}
            value={input.city}
          />
        </SchoolFormField>
        <SchoolFormField
          label="State"
          name="state"
          component={Select}
          options={states}
          onChange={({ option }) => setInput({ ...input, state: option })}
          value={input.state}
          placeholder="State"
        />
        <SchoolFormField label="Zip Code">
          <SchoolBaseTextInput
            name="zip"
            placeholder="90210"
            onChange={handleChanges}
            value={input.zip}
          />
        </SchoolFormField>
        <SchoolFormField label="Phone Number">
          <SchoolMaskedInput
            mask={[
              { fixed: '(' },
              {
                length: 3,
                regexp: /^[0-9]{1,3}$/,
                placeholder: 'xxx'
              },
              { fixed: ')' },
              { fixed: ' ' },
              {
                length: 3,
                regexp: /^[0-9]{1,3}$/,
                placeholder: 'xxx'
              },
              { fixed: '-' },
              {
                length: 4,
                regexp: /^[0-9]{1,4}$/,
                placeholder: 'xxxx'
              }
            ]}
            value={input.phone}
            name="phone"
            onChange={handleChanges}
            required
          />
        </SchoolFormField>
        <SchoolFormField label="Type of Institution">
          <SchoolBaseTextInput
            name="type"
            placeholder="University"
            onChange={handleChanges}
            value={input.type}
          />
        </SchoolFormField>
        <SchoolFormField label="Institution Website">
          <SchoolBaseTextInput
            labelText="Institution Website"
            name="url"
            placeholder="ls.dev"
            onChange={handleChanges}
            value={input.url}
            required
          />
        </SchoolFormField>
        <SchoolButton type="submit" primary label="Submit" alignSelf="center" />
      </Box>
    </SchoolForm>
  );
};

SchoolDetailsForm.propTypes = {
  history: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

const SchoolForm = styled(BaseForm)`
  margin: 120px auto 100px;
  background-color: white;
  border: 1px solid #d8d8d8;
  border-radius: 2px;
  max-width: 800px;
  width: 100%;
`;

const SchoolButton = styled(BaseButton)`
  text-align: center;
  margin: 15px 20px 50px;
`;

const SchoolBaseTextInput = styled(BaseTextInput)`
  border: none;
`;

const SchoolFormField = styled(BaseFormField)`
  margin: 20px;
  border-bottom: none;
  input {
    /* margin-bottom: 10px; */
  }
`;

const SchoolMaskedInput = styled(MaskedInput)`
  /* border: ${({ theme }) => theme.global.border}; */
`;

export default SchoolDetailsForm;
