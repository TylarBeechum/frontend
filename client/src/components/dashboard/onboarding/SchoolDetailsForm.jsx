import React, { useState } from 'react';

import { addSchoolDetails } from './onboardQueries';

const SchoolDetailsForm = ({ history }) => {
  const [input, setInput] = useState({
    name: '',
    taxId: '',
    street1: '',
    street2: '',
    city: '',
    state: '',
    zip: '',
    phone: '',
    type: '',
    url: '',
    userId: 1
  });

  const handleChanges = e => {
    setInput({
      ...input,
      [e.target.name]: e.target.value
    });
  };

  const handleSubmit = async e => {
    e.preventDefault();
    try {
      await addSchoolDetails(input);
    } catch (err) {
      console.log(err);
    }

    history.push('/dashboard');
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        name="name"
        placeholder={'Name of Institution'}
        onChange={handleChanges}
        value={input.name}
      />
      <input
        name="taxId"
        placeholder={'TaxId'}
        onChange={handleChanges}
        value={input.taxId}
      />
      <input
        name="street1"
        placeholder={'Address 1'}
        onChange={handleChanges}
        value={input.street1}
      />
      <input
        name="street2"
        placeholder={'Address 2'}
        onChange={handleChanges}
        value={input.street2}
      />
      <input
        name="city"
        placeholder={'City'}
        onChange={handleChanges}
        value={input.city}
      />
      <input
        name="state"
        placeholder={'State'}
        onChange={handleChanges}
        value={input.state}
      />
      <input
        name="zip"
        placeholder={'Zip Code'}
        onChange={handleChanges}
        value={input.zip}
      />
      <input
        name="phone"
        placeholder={'Phone Number'}
        onChange={handleChanges}
        value={input.phone}
      />
      <input
        name="type"
        placeholder={'Type of Institution'}
        onChange={handleChanges}
        value={input.type}
      />
      <input
        name="url"
        placeholder={'Institution Website'}
        onChange={handleChanges}
        value={input.url}
      />
      <button type="submit">Submit</button>
    </form>
  );
};

export default SchoolDetailsForm;
