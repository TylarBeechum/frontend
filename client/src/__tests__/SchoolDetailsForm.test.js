/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import {
  renderWithRouterAndProviders,
  rendererWithRouterAndProviders
} from '../__testHelpers__';

import SchoolDetailsForm from '../components/Dashboard/Onboard/SchoolDetailsForm';

describe('<SchoolDetailsForm />', () => {
  describe('should render the component successfully without crashing', () => {
    it('tests that the component does not crash when it is rendered', () => {
      const helpers = renderWithRouterAndProviders(
        <SchoolDetailsForm history={{ location: { pathname: '/' } }} />
      );
    });
  });

  describe('should match the previous snapshot of the component', () => {
    it('tests that the component has not changed and matched the previous snapshot', () => {
      const tree = rendererWithRouterAndProviders(
        <SchoolDetailsForm history={{ location: { pathname: '/' } }} />
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe('should render headers and labels to the screen successfully', () => {
    it('tests that the header test renders correctly', () => {
      const { getByText } = renderWithRouterAndProviders(
        <SchoolDetailsForm history={{ location: { pathname: '/' } }} />
      );
      expect(getByText(/School Register/i)).toBeDefined();
    });

    it('tests the Institution label renders correctly', () => {
      const { getByText } = renderWithRouterAndProviders(
        <SchoolDetailsForm history={{ location: { pathname: '/' } }} />
      );
      expect(getByText(/Institution/i)).toBeDefined();
    });

    it('tests the Institution label renders correctly', () => {
      const { getAllByText } = renderWithRouterAndProviders(
        <SchoolDetailsForm history={{ location: { pathname: '/' } }} />
      );
      expect(getAllByText(/Institution/i)).toBeDefined();
    });
  });
});
