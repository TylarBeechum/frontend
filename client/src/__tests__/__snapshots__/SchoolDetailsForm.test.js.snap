// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SchoolDetailsForm /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 iDwiuh"
>
  <form
    className="sc-htpNat sc-EHOje hyxSzg"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="StyledBox-sc-13pk1d4-0 IRSNj"
    >
      <h1
        className="StyledHeading-sc-1rdh4aw-0 ePLjJY"
      >
        School Register
      </h1>
      <div
        className="sc-ifAKCX sc-htoDjs NRMgl FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
      >
        <label
          className="StyledText-sc-1sadyjn-0 bKLvLm"
        >
          Institution
        </label>
        <div
          className="StyledBox-sc-13pk1d4-0 gZSidW"
        >
          <div
            className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
          >
            <input
              autoComplete="off"
              className="sc-bxivhb sc-gzVnrw bBtqGg StyledTextInput-sc-1x30a0s-0 duvNfq"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Lambda School"
              required={true}
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="sc-ifAKCX sc-htoDjs NRMgl FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
      >
        <label
          className="StyledText-sc-1sadyjn-0 bKLvLm"
        >
          Tax Id
        </label>
        <div
          className="StyledBox-sc-13pk1d4-0 gZSidW"
        >
          <div
            className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
          >
            <input
              autoComplete="off"
              className="sc-bxivhb sc-gzVnrw bBtqGg StyledTextInput-sc-1x30a0s-0 duvNfq"
              name="taxId"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="TaxId"
              required={true}
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="sc-ifAKCX sc-htoDjs NRMgl FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
      >
        <label
          className="StyledText-sc-1sadyjn-0 bKLvLm"
        >
          Address 1
        </label>
        <div
          className="StyledBox-sc-13pk1d4-0 gZSidW"
        >
          <div
            className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
          >
            <input
              autoComplete="off"
              className="sc-bxivhb sc-gzVnrw bBtqGg StyledTextInput-sc-1x30a0s-0 duvNfq"
              name="street1"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="123 Fake street"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="sc-ifAKCX sc-htoDjs NRMgl FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
      >
        <label
          className="StyledText-sc-1sadyjn-0 bKLvLm"
        >
          Address 2
        </label>
        <div
          className="StyledBox-sc-13pk1d4-0 gZSidW"
        >
          <div
            className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
          >
            <input
              autoComplete="off"
              className="sc-bxivhb sc-gzVnrw bBtqGg StyledTextInput-sc-1x30a0s-0 duvNfq"
              name="street2"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Apt B"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="sc-ifAKCX sc-htoDjs NRMgl FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
      >
        <label
          className="StyledText-sc-1sadyjn-0 bKLvLm"
        >
          City
        </label>
        <div
          className="StyledBox-sc-13pk1d4-0 gZSidW"
        >
          <div
            className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
          >
            <input
              autoComplete="off"
              className="sc-bxivhb sc-gzVnrw bBtqGg StyledTextInput-sc-1x30a0s-0 duvNfq"
              name="city"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="San Francisco"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="sc-ifAKCX sc-htoDjs NRMgl FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
      >
        <label
          className="StyledText-sc-1sadyjn-0 bKLvLm"
        >
          State
        </label>
        <div
          className="StyledBox-sc-13pk1d4-0 gZSidW"
        >
          <button
            aria-label="Open Drop"
            className="Select__StyledSelectDropButton-sc-17idtfo-1 cvpccN StyledButton-sc-323bzc-0 jfbKsv"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            <div
              className="StyledBox-sc-13pk1d4-0 erazuK"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 XINUX"
              >
                <div
                  className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
                >
                  <input
                    autoComplete="off"
                    className="Select__SelectTextInput-sc-17idtfo-0 lmyIUy StyledTextInput-sc-1x30a0s-0 duvNfq"
                    label="State"
                    name="state"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="State"
                    readOnly={true}
                    tabIndex="-1"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="StyledBox-sc-13pk1d4-0 eLQtUu"
                style={
                  Object {
                    "minWidth": "auto",
                  }
                }
              >
                <svg
                  aria-label="FormDown"
                  className="StyledIcon-ofa7kd-0 dSllCE"
                  viewBox="0 0 24 24"
                >
                  <polyline
                    fill="none"
                    points="18 9 12 15 6 9"
                    stroke="#000"
                    strokeWidth="2"
                  />
                </svg>
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        className="sc-ifAKCX sc-htoDjs NRMgl FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
      >
        <label
          className="StyledText-sc-1sadyjn-0 bKLvLm"
        >
          Zip Code
        </label>
        <div
          className="StyledBox-sc-13pk1d4-0 gZSidW"
        >
          <div
            className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
          >
            <input
              autoComplete="off"
              className="sc-bxivhb sc-gzVnrw bBtqGg StyledTextInput-sc-1x30a0s-0 duvNfq"
              name="zip"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="90210"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="sc-ifAKCX sc-htoDjs NRMgl FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
      >
        <label
          className="StyledText-sc-1sadyjn-0 bKLvLm"
        >
          Phone Number
        </label>
        <div
          className="StyledBox-sc-13pk1d4-0 gZSidW"
        >
          <div
            className="StyledMaskedInput__StyledMaskedInputContainer-sc-99vkfa-1 kyyRCm"
          >
            <input
              autoComplete="off"
              className="sc-dnqmqq djyQej StyledMaskedInput-sc-99vkfa-0 kedzFm"
              name="phone"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="(xxx) xxx-xxxx"
              required={true}
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="sc-ifAKCX sc-htoDjs NRMgl FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
      >
        <label
          className="StyledText-sc-1sadyjn-0 bKLvLm"
        >
          Type of Institution
        </label>
        <div
          className="StyledBox-sc-13pk1d4-0 gZSidW"
        >
          <div
            className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
          >
            <input
              autoComplete="off"
              className="sc-bxivhb sc-gzVnrw bBtqGg StyledTextInput-sc-1x30a0s-0 duvNfq"
              name="type"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="University"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="sc-ifAKCX sc-htoDjs NRMgl FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
      >
        <label
          className="StyledText-sc-1sadyjn-0 bKLvLm"
        >
          Institution Website
        </label>
        <div
          className="StyledBox-sc-13pk1d4-0 gZSidW"
        >
          <div
            className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
          >
            <input
              autoComplete="off"
              className="sc-bxivhb sc-gzVnrw bBtqGg StyledTextInput-sc-1x30a0s-0 duvNfq"
              name="url"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="ls.dev"
              required={true}
              value=""
            />
          </div>
        </div>
      </div>
      <button
        className="sc-bdVaJa sc-bZQynM iMOZtP StyledButton-sc-323bzc-0 jLEySZ"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
